c2rust-transpile/src/c_ast/conversion.rs

```diff
@@ -240,22 +240,16 @@ pub struct ConversionContext {
 fn display_loc(ctx: &AstContext, loc: &Option<SrcSpan>) -> Option<DisplaySrcSpan> {
     loc.as_ref().map(|loc| DisplaySrcSpan {
         file: ctx.files[loc.fileid as usize].path.clone(),
-        loc: loc.clone(),
+        loc: *loc,
     })
 }
 
 fn has_packed_attribute(attrs: Vec<Value>) -> bool {
-    for attr in attrs {
-        match from_value::<String>(attr.clone())
-            .expect("Record attributes should be strings")
-            .as_str()
-        {
-            "packed" => return true,
-            _ => {}
-        }
-    }
-
-    return false;
+    attrs.iter().any(|attr| {
+        let attr_name: String =
+            from_value(attr.clone()).expect("Record attributes should be strings");
+        attr_name == "packed"
+    })
```

Comment:
Maybe? I don't know. I have focused on fixing the lints, performing equivalent code transformations, and this change requires thinking long and hard about how all code fits together. Since it was good enough before the change, it should be good enough after the change as well.

Overall, the codebase is too panic-happy, and would benefit from a cleanup of error handling, but that's a separate issue.

Reply:
Panicking if this is not a `String` is a consistency check between the CBOR generator and consumer, and should stay in.

Reply:
Ok, we'll leave it.  It would make sense to fix it in a separate PR if we did want to change it, but I wanted to mark it here before it gets lost.